partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedNestedKeyPartitoner.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.storage.partitioner;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.data.Timestamp;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.DataException;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.Map;
+
+import io.confluent.connect.storage.errors.PartitionException;
+
+public class TimeBasedNestedKeyPartitoner<T> extends TimeBasedPartitioner<T> {
```

Comment:
We've forked kafka-connect-storage-common here, but should the new partitioner be it's own project that we can push up on to artifactory, then have an individual partitioner jar that we can drop into any connector docker container?

Reply:
As opposed to just building the whole connector as a jar?

Reply:
Yes, the partitioner will have its own jar. Sad we will have to update the confluent deployment because this is a different jar from the connector jar.

Reply:
At least if we make the partitioner its own jar, we don't have to maintain our own version of kafka connect. @johnnycaol Is it correct that we can freely update connect every time confluent makes changes and easily plug our partitioner in with 0 code changes needed by us?

Reply:
@deysman currently, I have my own logic implemented in its own class which extends from the base class. So I can't say 0 code change as if they change the behaviour of the base class, I might have to change my class as well. However, the change should be minimum. I'm not sure about the how we gonna deal the customized partitioner for other connectors, if we want to share the same partitioner across the board, then it makes sense to update the confluent deployment to use our own partitioner. Depends on the requirements.

Reply:
@johnnycaol We could rip out your code that extends the partitioner to be completely separate from kafka connect. Then in the s3 connector config we point it to your implementation with partitioner.class = com.flipp.johhnys.sweet.partitioner. This way instead of replacing kafka-connect-storage-partitioner-3.3.0.jar in the share folder we can add an additional jar.

Reply:
And anyone who wants to use your partitioner just drops your new jar file into that folder, no replacing confluent code necessary.

Reply:
If we can build the partitioner as its own jar and simply include it that way, I would prefer to do that.